functions-iterators/starter.js

```diff
@@ -1,82 +1,236 @@
-/* ============ BEFORE YOU GET STARTED ============ */
+// //PART0://
+var name = "Jumanah"
 
-/* here is a very short function that pops up an alert window that says 'Hello!' */
-function hello() {
-  alert('Hello!');
+function justDoIt(name) {
+    return name + " " + "JUST DO IT !!";
 }
 
-/* Here's the steps to test it:
+// console.log(justDoIt(name));
 
-- Open index.html in chrome.
-- Type the keyboard shortcut `cmd + option + j`. Dev tools should open to the console.
-- You should see the message saying 'starter.js is connected!'
-- Underneath that, next to the blue arrow, type `hello()` to call the `hello` function
-- You should get an alert window that says HELLO!
 
-*/
+// //PART1://
 
-/* ================= AS YOU COMPLETE EACH STEP ============= */
 
-/* After each step, perform the following commands in the terminal:
-1. git add .
-2. git commit -m "add completed part PART-NUMBER-HERE"
-This will help us track your progress and it will give you practice writing good commit messages!! 
-*/
+function bigOrSmallString(bigOrS) {
+    if (bigOrS.length >= 10) {
+
+        return (`${bigOrS} "This word is loooooong!"`);
+    } else {
+        return (`${bigOrS}"this is short"`);
+    }
+}
+//  console.log(bigOrSmallString('hlolalala'));
 
-/* ++++====+++==++++==+++++======++++===++++++=====+++==++++++++++===== */
-/* ======================= WRITE YOUR ANSWERS HERE!!!!! =============== */
-/* ++++====+++==++++==+++++======++++===++++++=====+++==++++++++++===== */
 
-// PART 0: Just DO it!!
 
+// // PART2 //
 
 
-// PART 1: Big or Small String?
 
+function oddOrEvenString(wordLength) {
+    if (wordLength.length() % 2 === 0) {
+        return ("This string length is TOTALLY even!")
+    } else {
+        return ("This string's length is odd..")
+    }
 
+}
 
-// PART 2: Odd or Even String Length?
+//console.log(oddOrEvenString('complicated')); 
 
 
 
-// PART 3: Median
+// //PART3 //
 
 
+var array = [1, 5, 6, 2, 3, 4];
 
-// PART 4: Sum Array
+function medianOfArray(array) {
+    array.sort();
+    if (array.length % 2 === 0) {
+        var array2 = array.length / 2;
+        array[array2];
 
 
+    } else {
+        var array3 = array.lenth / 2;
+        var jay = array[array3] + array[array3 + 1];
+        return jay;
+    }
+}
 
-// PART 5: Vowel Count
 
 
 
-// PART 6: Initials
+// // PART 4 //
 
 
+var numbers = [1, 2, 5, 6, 7];
+var sum = 0;
+function sumArray(numbers) {
+    for (var i = 0; i < numbers.length; i++) {
+        sum += numbers[i]
+    }
+}
+//  console.log(numbers);
```

Comment:
you can use reduce method, but if you want to do it like this you need first to call the function first  `sumArray(numbers)` so it can do the sum then log the sum value `console.log(sum);`